js/woo.js

```diff
@@ -0,0 +1,1465 @@
+            const currencyCode = currencyItem['code'];
+            if (chainedCode === currencyCode) {
+                return chainedCode;
+            } else {
+                const networks = currencyItem['networks'];
+                const networkKeys = Object.keys (networks);
+                for (let j = 0; j < networkKeys.length; j++) {
+                    const networkKey = networkKeys[j];
+                    const networkItem = networks[networkKey];
+                    if (networkItem['chained_currency_code'] === chainedCode) {
+                        return currencyItem;
+                    }
+                }
+            }
+        }
+        return undefined;
+    }
+
+    objectMemberAt (objOrArray, position) { // TODO: can be in base
+        return objOrArray[Object.keys (objOrArray)[position]];
+    }
+
+    hasChildWithKeyValue (obj, targetKey, targetValue) {
+        const keys = Object.keys (obj);
+        for (let i = 0; i < keys.length; i++) {
+            const currentKey = keys[i];
+            const childMember = obj[currentKey];
+            const value = this.safeInteger (childMember, targetKey, undefined);
+            if (value === targetValue) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    sanitizestr (str) { // TODO: this method can be moved into base
+        str = str.replace (' ', '-');
+        str = str.replace ('.', '-');
+        str = str.replace ('_', '-');
+        return str.toLowerCase ();
+    }
+
+    detectChainSlug (type) { // TODO: this method can be moved into base
+        const type_S = this.sanitizestr (type);
+        const chainsSlugs = {};
+        chainsSlugs['BTC'] = [ 'btc', 'btc-chain', 'btc-network', 'bitcoin', 'bitcoin-chain', 'bitcoin-network' ];
+        chainsSlugs['BEP20'] = [ 'bep', 'bep20', 'bep-20', 'bsc', 'bsc20', 'bsc-20', 'binance', 'binance-network', 'binance-smart-chain', 'bep20-bsc' ]; // i.e. lbank has 'bep20(bsc)'
+        chainsSlugs['BEP2'] = [ 'bep2', 'bep-2', 'bnb', 'binance-chain', 'binance-network' ];
+        chainsSlugs['ERC20'] = [ 'erc', 'erc20', 'erc-20', 'eth', 'eth20', 'eth-20', 'ethereum', 'ethereum-network', 'ethereum-chain' ];
+        chainsSlugs['HRC20'] = [ 'heco', 'hrc', 'hrc20', 'hrc-20', 'huobi', 'huobi-network', 'huobi-chain', 'huobi-eco-chain', 'eco-network', 'eco-chain' ];
+        chainsSlugs['TRC20'] = [ 'trc', 'trc20', 'trc-20', 'trx', 'trx20', 'trx-20', 'tron', 'tron-network', 'tron-chain', 'trx-chain', 'trx-network' ];
+        chainsSlugs['SOL'] = [ 'sol', 'solana', 'solana-network', 'solana-chain', 'sol-network', 'sol-chain' ];
+        chainsSlugs['MATIC'] = [ 'matic', 'matic-network', 'matic-chain', 'polygon', 'polygon-network', 'polygon-chain' ];
+        const keys = Object.keys (chainsSlugs);
+        for (let i = 0; i < keys.length; i++) {
+            const key = keys[i];
+            const chainList = chainsSlugs[key];
+            if (chainList.includes (type_S)) {
```

Comment:
We don't transpile `includes`, cause a linear search in the array is always slower than a key-lookup in a dictionary, which is why we never use loops and arrays for such things, instead we use dictionaries with safe-methods.